flink-runtime/src/main/java/org/apache/flink/streaming/api/windowing/assigners/SlidingEventTimeWindows.java

```diff
@@ -59,6 +59,11 @@ protected SlidingEventTimeWindows(long size, long slide, long offset) {
                     "SlidingEventTimeWindows parameters must satisfy "
                             + "abs(offset) < slide and size > 0");
         }
+        if (size / slide > 10000000) {
```

Comment:
Why this particular number? Suggestion: use a constant, so not to use magic numbers in the code.

Reply:
I test on my local environment, OOM occurs if it is bigger than 2000000+, so I set 1000000 here.
I want introduce a config, but there is no chance to visit `Configuration`.

Reply:
Why can't we introduce a config for this one? Like @nilmadhab I am nervous about hard coding a number in the code that may not be always appropriate. It was be useful to understand how we would advise on this number.
- Is it heap related?  
- I wonder what the limits of this use case for this API are, ideally from stakeholders point of view. Rather than setting something very large that no-one would reasonably use.

Reply:
I have already added the config.

- It is related the JVM heap.
- `SlidingEventTimeWindows` is an open API users could set the parameters of the constructor arbitrarily.

Reply:
> Why this particular number? Suggestion: use a constant, so not to use magic numbers in the code.

+1 for using the Constant instead of magic number.